functions/utils/index.js

```diff
@@ -12,26 +21,52 @@ const getQueryKeys = (properties) =>
         value,
       } = property;
 
-      if (kind === 'BELONGS_TO' && typeof value === 'number') {
-        return `${name} {
-          id\n
-        }`;
-      }
+      switch (true) {
+        case kind === RelationKind.BELONGS_TO && typeof value === 'number':
+        case kind === RelationKind.HAS_AND_BELONGS_TO_MANY &&
+          isNumberArray(value):
+        case kind === RelationKind.HAS_MANY && isNumberArray(value):
+          return `${name} {
+            id\n
+          }`;
 
-      if (kind === 'BELONGS_TO' && isRecord(value)) {
-        const keys = Object.keys(value);
+        case kind === RelationKind.BELONGS_TO && isRecord(value): {
+          const keys = Object.keys(value);
+          return `${name} {
+            ${keys.map((key) => key).join('\n')}
+          }`;
+        }
 
-        return `${name} {
-          ${keys.map((key) => key).join('\n')}
-        }`;
-      }
+        case kind === RelationKind.HAS_AND_BELONGS_TO_MANY &&
+          isCollection(value):
+        case kind === RelationKind.HAS_MANY && isCollection(value): {
+          const keys = value[0] ? Object.keys(value[0]) : ['id'];
+          return `${name} {
+            ${keys.map((key) => key).join('\n')}
+          }`;
+        }
 
-      return name;
+        default:
+          return name;
+      }
```

Comment:
I'm not sure about this `switch` statement. What do you think about something like this 
```
let output = '';

switch(kind) {
  case RelationKind.BELONGS_TO:
    output = parseBelongsTo(value)
  case RelationKind.HAS_MANY:
    output = parseHasMany(value)
  case RelationKind.HAS_AND_BELONGS_TO_MANY:
    output = parseHasAndBelongsToMany(value)
  default:
    output = name;
}

return output;
```